packages/kbn-apm-synthtrace/src/lib/apm/instance.ts

```diff
@@ -22,6 +23,45 @@ export class Instance extends Entity<ApmFields> {
     });
   }
 
+  dbExitSpan({ spanName, subType }: { spanName: string; subType?: string }) {
+    const exitSpan = new Span({
+      ...this.fields,
+      'service.target.type': subType,
+      'span.destination.service.name': '', //deprecated
+      'span.destination.service.resource': subType,
+      'span.destination.service.type': '', //deprecated
+      'span.name': spanName,
+      'span.subtype': subType,
+      'span.type': 'db',
+    });
+
+    return exitSpan;
+  }
+
+  messagingExitSpan(mesageQueueName: string) {}
+
+  httpExitSpan({ spanName, destination }: { spanName: string; destination: string }) {
+    // host: 'opbeans-go:3000',
+    // hostname: 'opbeans-go',
+    // port: '3000',
+    const parsed = new url.URL(destination);
+
+    const exitSpan = new Span({
+      ...this.fields,
+      'destination.address': parsed.hostname,
+      'destination.port': parseInt(parsed.port, 10),
+      'service.target.name': parsed.host,
+      'span.destination.service.name': '', //deprecated
```

Comment:
isn't `.resource` deprecated instead? and I think they should all be set, in any case?